src/test/java/io/confluent/examples/streams/IntegrationTestUtils.java

```diff
@@ -296,24 +315,37 @@ private static boolean continueConsuming(final int messagesConsumed, final int m
     return results;
   }
 
-  static <K, V> void produceKeyValuesSynchronously(final String inputTopic,
-                                                   final List<KeyValue<K, V>> entities,
+  /**
+   * Like {@link IntegrationTestUtils#produceKeyValuesSynchronously(String, Collection, Properties)}, except for use
+   * with TopologyTestDriver tests, rather than "native" Kafka broker tests.
+   *
+   * @param topic              Kafka topic to write the data records to
+   * @param records            Data records to write to Kafka
+   * @param topologyTestDriver The {@link TopologyTestDriver} to send the data records to
+   * @param keySerializer      The {@link Serializer} corresponding to the key type
+   * @param valueSerializer    The {@link Serializer} corresponding to the value type
+   * @param <K>                Key type of the data records
+   * @param <V>                Value type of the data records
+   */
+  static <K, V> void produceKeyValuesSynchronously(final String topic,
+                                                   final List<KeyValue<K, V>> records,
                                                    final TopologyTestDriver topologyTestDriver,
                                                    final Serializer<K> keySerializer,
                                                    final Serializer<V> valueSerializer) {
-    for (final KeyValue<K, V> entity : entities) {
-      topologyTestDriver.pipeInput(new ConsumerRecord<>(
-        inputTopic,
+    for (final KeyValue<K, V> entity : records) {
+      final ConsumerRecord<byte[], byte[]> consumerRecord = new ConsumerRecord<>(
+        topic,
         0,
         0,
         0,
```

Comment:
If we set timestamp to zero for all, this might be bad for testing windows. Should we pass in a timestamp per key-value pair, too? Or similar? Or use `RecordFactory` ?

Reply:
I figured I'd cross that bridge when I came to a test that needs it. Is it ok with you to leave this as-is for now, and then implement whatever timestamp support is appropriate when we come to a test that needs to verify windows?

Reply:
Fine with me.